File: gcc/testsuite/gcc.target/riscv/weak-1.c
/* { dg-do compile } */
/* { dg-options "-mcmodel=medany -mexplicit-relocs -O" } */

/* Verify that the branch doesn't get optimized away.  */
extern int weak_func(void) __attribute__ ((weak));

int
sub (void)
{
  if (weak_func)
    return weak_func ();
  return 0;
}
/* { dg-final { scan-assembler "b\(ne|eq\)" } } */

File: gcc/testsuite/gfortran.dg/weak-1.f90
! { dg-do compile }
! { dg-require-weak "" }
! { dg-final { scan-assembler "weak\[^ \t\]*\[ \t\]_?impl" { target { ! nvptx-*-* } } } }
! { dg-final { scan-assembler-times "\\.weak \\.func impl" 2 { target nvptx-*-* } } }
subroutine impl
!GCC$ ATTRIBUTES weak :: impl
end subroutine
